### ISSUE TITLE
Trying to view node settings/fields results in "Unknown error" flashing up in toast

### ISSUE THREAD

**Issue by davebeesley** [verbb/navigation#124]
### Description
Trying to view node settings/fields results in "Unknown error" flashing up in toast

Error Log
```
2020-01-09 11:24:28 [-][1][-][error][Twig\Error\SyntaxError] Twig\Error\SyntaxError: Unknown "displayName" function. in /var/www/html/vendor/verbb/navigation/src/templates/navs/_editor.html:10
Stack trace:
#0 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(451): Twig\ExpressionParser->getFunctionNodeClass('displayName', 10)
#1 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(235): Twig\ExpressionParser->getFunctionNode('displayName', 10)
#2 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(175): Twig\ExpressionParser->parsePrimaryExpression()
#3 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(70): Twig\ExpressionParser->getPrimary()
#4 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(380): Twig\ExpressionParser->parseExpression()
#5 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(281): Twig\ExpressionParser->parseHashExpression()
#6 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(175): Twig\ExpressionParser->parsePrimaryExpression()
#7 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(70): Twig\ExpressionParser->getPrimary()
#8 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(338): Twig\ExpressionParser->parseExpression()
#9 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(279): Twig\ExpressionParser->parseArrayExpression()
#10 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(175): Twig\ExpressionParser->parsePrimaryExpression()
#11 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(70): Twig\ExpressionParser->getPrimary()
#12 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(593): Twig\ExpressionParser->parseExpression(0, true)
#13 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(551): Twig\ExpressionParser->parseArguments(true, false, true)
#14 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(539): Twig\ExpressionParser->parseFilterExpressionRaw(Object(Twig\Node\Expression\NameExpression))
#15 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(397): Twig\ExpressionParser->parseFilterExpression(Object(Twig\Node\Expression\NameExpression))
#16 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(289): Twig\ExpressionParser->parsePostfixExpression(Object(Twig\Node\Expression\NameExpression))
#17 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(175): Twig\ExpressionParser->parsePrimaryExpression()
#18 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(70): Twig\ExpressionParser->getPrimary()
#19 /var/www/html/vendor/twig/twig/src/ExpressionParser.php(663): Twig\ExpressionParser->parseExpression()
#20 /var/www/html/vendor/twig/twig/src/TokenParser/SetTokenParser.php(38): Twig\ExpressionParser->parseMultitargetExpression()
#21 /var/www/html/vendor/twig/twig/src/Parser.php(185): Twig\TokenParser\SetTokenParser->parse(Object(Twig\Token))
#22 /var/www/html/vendor/twig/twig/src/TokenParser/ForTokenParser.php(52): Twig\Parser->subparse(Array)
#23 /var/www/html/vendor/twig/twig/src/Parser.php(185): Twig\TokenParser\ForTokenParser->parse(Object(Twig\Token))
#24 /var/www/html/vendor/twig/twig/src/Parser.php(98): Twig\Parser->subparse(NULL, false)
#25 /var/www/html/vendor/twig/twig/src/Environment.php(563): Twig\Parser->parse(Object(Twig\TokenStream))
#26 /var/www/html/vendor/twig/twig/src/Environment.php(595): Twig\Environment->parse(Object(Twig\TokenStream))
#27 /var/www/html/vendor/craftcms/cms/src/web/twig/Environment.php(63): Twig\Environment->compileSource(Object(Twig\Source))
#28 /var/www/html/vendor/twig/twig/src/Environment.php(408): craft\web\twig\Environment->compileSource(Object(Twig\Source))
#29 /var/www/html/vendor/twig/twig/src/Environment.php(381): Twig\Environment->loadClass('__TwigTemplate_...', 'navigation/navs...', NULL)
#30 /var/www/html/vendor/craftcms/cms/src/web/twig/Environment.php(44): Twig\Environment->loadTemplate('navigation/navs...', NULL)
#31 /var/www/html/vendor/twig/twig/src/Environment.php(359): craft\web\twig\Environment->loadTemplate('navigation/navs...')
#32 /var/www/html/vendor/twig/twig/src/Environment.php(318): Twig\Environment->load('navigation/navs...')
#33 /var/www/html/vendor/craftcms/cms/src/web/View.php(344): Twig\Environment->render('navigation/navs...', Array)
#34 /var/www/html/vendor/verbb/navigation/src/controllers/NodesController.php(107): craft\web\View->renderTemplate('navigation/navs...', Array)
#35 [internal function]: verbb\navigation\controllers\NodesController->actionEditor()
#36 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#37 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#38 /var/www/html/vendor/craftcms/cms/src/web/Controller.php(187): yii\base\Controller->runAction('editor', Array)
#39 /var/www/html/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('editor', Array)
#40 /var/www/html/vendor/craftcms/cms/src/web/Application.php(299): yii\base\Module->runAction('navigation/node...', Array)
#41 /var/www/html/vendor/craftcms/cms/src/web/Application.php(565): craft\web\Application->runAction('navigation/node...', Array)
#42 /var/www/html/vendor/craftcms/cms/src/web/Application.php(278): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#43 /var/www/html/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#44 /var/www/html/web/index.php(21): yii\base\Application->run()
#45 {main}
```

### Steps to reproduce

1. Go to a Navigation group
2. Double click on a node (or click the cog next to it)
3. Spinner appears, then after a few seconds - a toast appears saying "An unknown error occurred."

### Additional info

- Plugin version: 1.2.1
- Craft version: Craft Pro 3.3.17
- PHP version: 7.4.0
- Linux: 4.9.184-linuxkit

**Comment by engram-design** [comment]
Fixed in [1.2.2](https://github.com/verbb/navigation/releases/tag/1.2.2). https://github.com/verbb/navigation/issues/122

**Comment by davebeesley** [comment]
Thanks for sorting that so quickly.
That fixed that, but now getting the following error on clicking save.

```
2020-01-09 11:58:03 [-][1][-][error][TypeError] TypeError: Argument 1 passed to craft\services\Elements::getElementById() must be of the type int, array given, called in /var/www/html/vendor/verbb/navigation/src/elements/Node.php on line 129 and defined in /var/www/html/vendor/craftcms/cms/src/services/Elements.php:288
Stack trace:
#0 /var/www/html/vendor/verbb/navigation/src/elements/Node.php(129): craft\services\Elements->getElementById(Array, 'craft\\elements\\...', 1)
#1 /var/www/html/vendor/verbb/navigation/src/elements/Node.php(214): verbb\navigation\elements\Node->getElement()
#2 /var/www/html/vendor/verbb/navigation/src/elements/Node.php(187): verbb\navigation\elements\Node->getElementUrl()
#3 /var/www/html/vendor/yiisoft/yii2/base/Component.php(139): verbb\navigation\elements\Node->getUrl()
#4 /var/www/html/vendor/craftcms/cms/src/base/Element.php(1004): yii\base\Component->__get('url')
#5 /var/www/html/vendor/verbb/navigation/src/controllers/NodesController.php(197): craft\base\Element->__get('url')
#6 /var/www/html/vendor/verbb/navigation/src/controllers/NodesController.php(24): verbb\navigation\controllers\NodesController->_setNodeFromPost()
#7 [internal function]: verbb\navigation\controllers\NodesController->actionSaveNode()
#8 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#9 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#10 /var/www/html/vendor/craftcms/cms/src/web/Controller.php(187): yii\base\Controller->runAction('save-node', Array)
#11 /var/www/html/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('save-node', Array)
#12 /var/www/html/vendor/craftcms/cms/src/web/Application.php(299): yii\base\Module->runAction('navigation/node...', Array)
#13 /var/www/html/vendor/craftcms/cms/src/web/Application.php(565): craft\web\Application->runAction('navigation/node...', Array)
#14 /var/www/html/vendor/craftcms/cms/src/web/Application.php(278): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#15 /var/www/html/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#16 /var/www/html/web/index.php(21): yii\base\Application->run()
#17 {main}
```

**Comment by engram-design** [comment]
Fixed in [1.2.3](https://github.com/verbb/navigation/releases/tag/1.2.3)

**Comment by davebeesley** [comment]
Awesome. Thanks
